### DISCUSSION TITLE
[Meta]: Static Site Generation

### DISCUSSION THREAD

**Discussion by pmelab** [wakujs/waku#691] [Ideas]
I would like to start a general brainstorming around what it would take to make Waku fill the Gatsby-shaped hole in the content-driven website world. I already did quite some experimentation and filed pull requests to resolve the most critical blockers, and the results look very promising. I will start with a run-down of the Gatsby features that are critical for our use case. Everybody interested in this topic is invited to join the discussion!

> **Disclaimer:** I don’t think there is a lot missing. I’m not recommending to turn Waku into Gatsby. I intend to collect use cases and discuss how they can be solved with the new tools RSC’s bring to the table.

* **Source plugins:** I think those can be largely replaced with data fetchers in server components.
* **Data store & cache**: If there is one thing that caused the most problems for us in years of building websites with Gatsby, it was the data store. Again, server components are the answer. Caching can be much better handled with domain knowledge in custom fetchers. This does not have to be implemented in the framework.
* **Incremental builds:** Gatsby’s incremental builds have two sides to them. There is incremental fetching into the datastore, and there is incremental building of HTML assets. The first part is irrelevant if data is fetched on demand in server components. For the latter I filed a PR. The assumption is that Waku will only care about files that are missing. It should not implement knowledge about what to delete before updating.
* **Image processing:** That’s also a case for server components.
* **Automatic host configuration:** Like generation of of `_headers`, `_redirects` (Netlify) and similar files. In my opinion not worth to be added to the framework since it can be handled easily outside.

## Related pull requests
* [x] #689 
* [x] #668 
* [x] #666 
* [x] #895 
* [ ] #928

**Comment by dai-shi** [comment]
Thanks for opening up a discussion!

> I would like to start a general brainstorming around what it would take to make Waku fill the Gatsby-shaped hole in the content-driven website world.

That would be wonderful! And, I'm pretty sure @sandren is looking in the same direction.
